src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxTriviaListExtensions.cs

```diff
@@ -90,5 +90,24 @@ public static IEnumerable<SyntaxTrivia> TakeRange(this SyntaxTriviaList triviaLi
                 yield return triviaList[start++];
             }
         }
+
+        /// <summary>
+        /// Returns modified <paramref name="triviaList"/> with removed multiple <see cref="SyntaxToken"/>
+        /// with kind <see cref="SyntaxKind.WhitespaceTrivia"/> arranged in a sequence 
+        /// preserving only the first one: [space1][space2][space3] -> [space1]. 
+        /// </summary>
+        public static SyntaxTriviaList CollapseSequentialWhitespaceTrivia(this SyntaxTriviaList triviaList)
```

Comment:
i don't think we really need this helper.

Reply:
Moved to a private method for now. Will remove once/if more beautiful solution will be found in separate "trivia handling" thread.